ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -1298,10 +1298,10 @@ impl LookupGlobalInst {
 pub struct CallInst {
     /// The callee.
     target: FuncDeclIdx,
-    /// The first argument to the call, if present. Undefined if not present.
-    arg1: PackedOperand,
-    /// Extra arguments, if the call requires more than a single argument.
-    extra: ExtraArgsIdx,
+    /// How many arguments in [Module::extra_args] is this call passing?
+    num_args: u16,
+    /// At what index do the contiguous operands in [Module::extra_args] start?
+    extra_idx: ExtraArgsIdx,
```

Comment:
This reminds me: Having tripped over the meaning of this field a couple of times in the past, I wanted to rename it to `first_extra_arg_idx` (or similar), but actually what you are proposing here opens us to even better naming.

If no call opcode has inline args any more, then in a sense every arg is "extra" and we may as well kill the "extra" terminology altogether.

So (maybe in another PR) we can rename anything `ExtraArg`-related to just `CallArg`? Then `extra_idx` in this strict becomes `first_arg_idx` (or similar)?

What do you think? Like I said, we don't have to do this now.

Reply:
I agree that this can be improved, and it can be done in another PR if we want.

Reply:
Oh one more thing that we may want to think about. For a non-vararg call we could even remove the `num_args` field as it's already known from the function signature.

I guess you knew that and decided to cache it here? That's fine, I don't mind much either way.

Reply:
Sorry, I've just seen that the next commit merges this code with vararg calls, so it makes complete sense to store the number of args.